ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -315,7 +319,15 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
         Transmission transmission = getTransmission(key);
         try {
           if (key.isConnectable()) {
-            handleConnect(key, transmission);
+            transmission.finishConnect();
+            if (transmission.ready()) {
+              connected.add(transmission.getConnectionId());
+              metrics.selectorConnectionCreated.inc();
+            } else {
+              pendingSslTrans.put(transmission.getConnectionId(),
```

Comment:
Well sure, they are connected but unready then. But then the ready check only makes sense for an established connection. The point is what is not ready is what you put in that data structure as per the code. I am just saying those do not necessarily have to be `SSLTransmission` as far as the selector is concerned.